homeassistant/components/bond/__init__.py

```diff
@@ -25,9 +25,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     host = entry.data[CONF_HOST]
     token = entry.data[CONF_ACCESS_TOKEN]
 
-    bond = Bond(bondIp=host, bondToken=token)
+    bond = Bond(host=host, token=token)
```

Comment:
Since the library is now using aiohttp, we'd like it to accept an optional aiohttp client session. We want all integrations that use aiohttp to use the home assistant aiohttp client session. It can be retrieved with our aiohttp client helper.

https://github.com/home-assistant/core/blob/3480fb69962e5eefaddf46d3756e9be3476add20/homeassistant/helpers/aiohttp_client.py#L32-L36

Reply:
Thanks, @MartinHjelmare. Which integration is good to look at as an example of doing that?

Reply:
Any integration that uses aiohttp should use some variant of the same logic for this. Here's one library example:
https://github.com/bachya/regenmaschine/blob/4855a67f696934297bd611c970b86ac4a606dc50/regenmaschine/client.py#L20-L28

Reply:
Hmm, that example actually looks like it has a bug, since it's globbing non keyword parameters in the client signature and we're not passing the session as a keyword when instantiating the client.

Reply:
We're using an older version of the library and the library has changed. That's why it looks weird now.

Reply:
I add the optional session to the library, and tried to use the session provided by HA but it does not work. Bond integration needs to make several HTTP calls in succession and it only works for the first one, then fails. It looks like it is a known open bug in aiohttp: https://github.com/aio-libs/aiohttp/issues/4549
I added sleep(1) between the API calls to verify the problem goes away - will watch this bug...